app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt

```diff
@@ -0,0 +1,181 @@
+                Box(modifier = Modifier.align(Alignment.BottomEnd)) {
+                  CircleWithIcon(Icons.Filled.Edit, MaterialTheme.colorScheme.primary)
+                }
+              }
+              Spacer(modifier = Modifier.height(66.dp))
+              CustomInputField(
+                  value = name,
+                  onValueChange = { newName ->
+                    if (newName.length <= maxUsernameLength) {
+                      name = newName
+                    }
+                  },
+                  label = "Name",
+                  placeholder = "Current name",
+                  supportingText = "Max $maxUsernameLength characters",
+                  trailingIcon = Icons.Filled.Clear,
+                  modifier = Modifier.testTag("editProfileNameInput"))
+              Spacer(modifier = Modifier.height(16.dp))
+              Row {
+                CustomInputField(
+                    value = description,
+                    onValueChange = { newDescription ->
+                      if (newDescription.length <= maxDescriptionLength) {
+                        description = newDescription
+                      }
+                    },
+                    label = "Description",
+                    placeholder = "Current description",
+                    supportingText = "Max $maxDescriptionLength characters",
+                    singleLine = false,
+                    trailingIcon = Icons.Filled.Clear,
+                    modifier = Modifier.testTag("editProfileDescriptionInput"))
+              }
+              Spacer(modifier = Modifier.height(120.dp))
+              Box(
+                  modifier =
+                      Modifier.border(1.dp, PrimaryGradientBrush, RoundedCornerShape(30.dp))
+                          .width(320.dp)
+                          .height(48.dp)) {
+                    Button(
+                        onClick = { /* Handle button click */},
+                        modifier = Modifier.fillMaxWidth().testTag("saveProfileButton"),
+                        colors =
+                            ButtonDefaults.buttonColors(containerColor = lightThemeBackground)) {
+                          Text(
+                              text = "Save",
+                              fontWeight = FontWeight.Bold,
+                              fontSize = 16.sp,
+                              color = MaterialTheme.colorScheme.primary)
+                        }
+                  }
+            }
+      })
+}
+
+@Composable
+fun CircleWithIcon(icon: ImageVector, backgroundColor: Color) {
+  Box(
+      modifier =
+          Modifier.size(32.dp)
+              .background(color = backgroundColor, shape = CircleShape)
+              .clickable { /* Handle click */}) {
+        Icon(
+            imageVector = icon,
+            contentDescription = "Edit",
+            tint = LightGray,
+            modifier = Modifier.padding(6.dp))
+      }
+}
```

Comment:
I've noticed that this composable will be of use in a future screen, Playlist Overview Screen. Adding it to the MainComponents file might be a good idea.